### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by pietruska** [otto802/quickarchiver#26]
Hi,

if I activate the add-on, my thunderbird (91.4.0) gets very slow. If I deactivate the add-on, everything works fast.

How could this problem be solved? This add-on is very helpfull!

Jan

**Comment by otto802** [comment]
Yes, in the current implementation this happens when working with large lists of emails. The trigger to fetch needed information for the displayed quickarchiver link is a eventlistener on the single viewed list entry - means the performance depends on the number of displayed emails in the window (=> not the actual number of emails in the folder). 

On my machine (when I not need to scroll too much) it does'nt really bothers me. However I still have this on my list of improvements (maybe implement some kind of caching here).

Otto

**Comment by pietruska** [comment]
Hi Otto,

in my case, it´s not only related to the number of mails in the list. In one account/folder (POP) I have about 2.000 mails, and it´s slow. In another folder (IMAP) it´s only about 50 and it´s slow, too.
As an idea, instead of an eventlistener to implement the option to only allocate when Thunderbird stars, or mails are downloaded.

Best,
Jan

**Comment by otto802** [comment]
Improved with release https://github.com/otto802/quickarchiver/releases/tag/quickarchiver-0.2.10

**Comment by pietruska** [comment]
Perfect! It´s fast again.
